src/components/ResultCard.tsx

```diff
@@ -31,10 +32,15 @@ const ResultCard = memo(
       <CardContent>
         <div className="grid grid-cols-12 items-center gap-2">
           <div className="col-end-1">
-            <Checkbox
-              data-cy={`card-${datasetUUID}-checkbox`}
-              checked={checked}
-              onChange={() => onCheckboxChange(datasetUUID)}
+            <FormControlLabel
+              control={
+                <Checkbox
+                  data-cy={`card-${datasetUUID}-checkbox`}
+                  checked={checked}
+                  onChange={() => onCheckboxChange(datasetUUID)}
+                />
+              }
+              label={datasetName}
```

Comment:
This is a nice fix, but we don't want to add a label next to the checkbox, especially not with the `datasetName` that you can already see on the dataset card. So I think we need to talk a bit more about what to do here. From a UX perspective, the nicest thing might be to make the entire card element clickable to set and unset the checkbox.